From c467480cc8b9fc74a6ca41132d9a092f040ff1b0 Mon Sep 17 00:00:00 2001
From: mkegelei <mkegelei@ulb.ac.be>
Date: Wed, 28 Apr 2021 17:16:12 +0200
Subject: [PATCH] fix rosbridge version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 48b96d3..1847383 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,7 @@ if (NOT MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 endif()
 
-find_package(console_bridge 0.3 REQUIRED)
+find_package(console_bridge 1.0 REQUIRED)
 include_directories(SYSTEM ${console_bridge_INCLUDE_DIRS})
 link_directories(${console_bridge_LIBRARY_DIRS})
 
