From 32fadb3f321a804db5988c3e0b2930218491f0f2 Mon Sep 17 00:00:00 2001
From: mkegelei <mkegelei@ulb.ac.be>
Date: Wed, 21 Oct 2020 18:32:20 +0200
Subject: [PATCH] Fix build flags for yocto

---
 src/cmake/ARGoSBuildFlags.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/ARGoSBuildFlags.cmake b/cmake/ARGoSBuildFlags.cmake
index 2e7a85b..578001c 100644
--- a/cmake/ARGoSBuildFlags.cmake
+++ b/cmake/ARGoSBuildFlags.cmake
@@ -16,14 +16,14 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 #
 # General compilation flags
 #
-set(CMAKE_C_FLAGS                  "${CMAKE_C_FLAGS} -Wall")
+set(CMAKE_C_FLAGS                  "${CMAKE_C_FLAGS} -Wall -ldl")
 if(ARGOS_BUILD_NATIVE)
   set(CMAKE_C_FLAGS                "${CMAKE_C_FLAGS} -mtune=native -march=native")
 endif(ARGOS_BUILD_NATIVE)
 set(CMAKE_C_FLAGS_RELEASE          "${CMAKE_C_FLAGS_RELEASE} -O2 -DNDEBUG")
 set(CMAKE_C_FLAGS_DEBUG            "${CMAKE_C_FLAGS_DEBUG} -ggdb3")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO   "${CMAKE_C_FLAGS_RELWITHDEBINFO} ${CMAKE_C_FLAGS_DEBUG} ${CMAKE_C_FLAGS_RELEASE}")
-set(CMAKE_CXX_FLAGS                "${CMAKE_CXX_FLAGS} -Wall")
+set(CMAKE_CXX_FLAGS                "${CMAKE_CXX_FLAGS} -Wall -ldl")
 if(ARGOS_BUILD_NATIVE)
   set(CMAKE_CXX_FLAGS              "${CMAKE_CXX_FLAGS} -mtune=native -march=native")
 endif(ARGOS_BUILD_NATIVE)
-- 
2.17.1

